database-migration/migrations/28_add_missing_databases.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023 Sorint.lab S.p.A.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package service is a package that provides methods for querying data
+
+package migrations
+
+import (
+	"context"
+	"fmt"
+
+	migrate "github.com/xakep666/mongo-migrate"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func init() {
+	err := migrate.Register(func(db *mongo.Database) error {
+		if err := addMissingDatabases(db); err != nil {
+			return err
+		}
+
+		return nil
+	}, func(db *mongo.Database) error {
+		return nil
+	})
+
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		return
+	}
+}
+
+func addMissingDatabases(client *mongo.Database) error {
+	collectionName := "hosts"
+
+	filter := bson.M{"$or": bson.A{
+		bson.M{
+			"$expr": bson.M{
+				"$gt": bson.A{
+					bson.M{
+						"$size": bson.M{
+							"$ifNull": bson.A{"$features.oracle.database.unretrievedDatabases", bson.A{}},
+						},
+					},
+					0,
+				},
+			},
+		},
+		bson.M{
+			"$expr": bson.M{
+				"$gt": bson.A{
+					bson.M{
+						"$size": bson.M{
+							"$ifNull": bson.A{"$features.oracle.database.unlistedRunningDatabases", bson.A{}},
+						},
+					},
+					0,
+				},
+			},
+		},
+	}}
```

Comment:
the condition doesn't filter all the documents, consider to use something like this:
```
filter := bson.D{
		{"$or",
			bson.A{
				bson.D{{"features.oracle.database.unlistedRunningDatabases", bson.D{{"$exists", true}}}},
				bson.D{{"features.oracle.database.unretrievedDatabases", bson.D{{"$exists", true}}}},
			},
		},
	}
```